ZlobinSV/lab_1/main.cpp

```diff
@@ -0,0 +1,78 @@
+#include <iostream>
+#include <string>
+using namespace std;
+
+int main()
+{
+    int i = 0, zam2, n = 0, k = 0, sum[100];
+    string zam1, str, word[100];
+    char a;
+
+    cout << "Enter the string: ";
+
+    getline(cin, str);
+
+    str += " ";
+
+    while (i < str.length() - 1) {
+        while (str[i] != ' ' && str[i] != ',' && str[i] != '.' && str[i] != '!' && str[i] != '?') {
+            word[n] += str[i];
+            i++;
+        }
+        n++;
+        while (str[i] == ' ' || str[i] == ',' || str[i] == '.' || str[i] == '!' || str[i] == '?') {
+            word[n] += str[i];
+            i++;
+        }
+        if (word[n].length()>=2) {
+            if ((word[n][0] == ',' || word[n][0] == '.' || word[n][0] == '!' || word[n][0] == '?') && (word[n][1] == ',' || word[n][1] == '.' || word[n][1] == '!' || word[n][1] == '?')) {
+                cout << "Error! The pinching mark is in the middle of a word or more than one pinning mark appears after / before the word.";
+                return 0;
+            }
+            if ((word[n][word[n].length() - 1] == ',' || word[n][word[n].length() - 1] == '.' || word[n][word[n].length() - 1] == '!' || word[n][word[n].length() - 1] == '?') && (word[n][word[n].length() - 2] == ',' || word[n][word[n].length() - 2] == '.' || word[n][word[n].length() - 2] == '!' || word[n][word[n].length() - 2] == '?')) {
+                cout << "Error! The pinching mark is in the middle of a word or more than one pinning mark appears after / before the word.";
+                return 0;
+            }
+        }
+        if (word[n] == "," || word[n] == "." || word[n] == "!" || word[n] == "?") {
+            cout << "Error! The pinching mark is in the middle of a word or more than one pinning mark appears after / before the word.";
+            return 0;
+        }
+
+        n++;
+    }
+    n--;
```

Comment:
It seems like n-- here is redundant and not obvious. It should represent the count of words, not the countOfWords - 1.
Tt's okay to use` operator <` instead of `<=` in for-loops below